test-it/ViaTests/UnflattenFromJSONEmptyArray.via

```diff
@@ -0,0 +1,34 @@
+define(MyVI dv(.VirtualInstrument (
+    Locals: c(
+        e(dv(a(.Int16 *) (7) ) arr1)
+        e(dv(a(.Int16 * *) ((7)) ) arr2)
+        e(dv(a(.Int16 * * *) (((7))) ) arr3)
+        e(dv(a(.Int16 * * * *) ((((7)))) ) arr4)
+        e(dv(a(.Int16 *) (7) ) arr5)
+        e(dv(a(.Int16 * *) ((7)) ) arr6)
+        e(dv(a(.Int16 * * *) (((7))) ) arr7)
+        e(dv(a(.Int16 * * * *) ((((7)))) ) arr8)
+        e(ErrorCluster err)
+        e(dv(a(.String *) ()) path)
+    )
+    clump(
+        UnflattenFromJSON("[]" arr1 path true false false err)
+        Printf("1: %z %z\n" arr1 err)
```

Comment:
I think this is ok here. But, it would be better to compare against expected value in the .via itself -- which will be tolerant for floating point also.

Reply:
I think that's valuable to do for floating point in specific but I disagree that it should be the hard and fast rule.

Not much use having a file that just says "true true true true true".

Reply:
And it might not even make sense for floating point... The print serialization of floating point is an algorithm that gives the smallest unambiguous string. The math functions operating on the same values in memory should result in the same string. If the algorithm operates on the same IEEE value in memory and gets different results that would be a bug...

Given that, there are definitely a class of tests where that makes more sense. Like checking the results of HTTP responses. The headers are variable over time so you would want logic to calculate a pass / fail. But for tests that don't rely on side-effects the extra logic in the tests doesn't seem to add much value...

Reply:
I agree with @rajsite.  I think it's actually better in general to have the expected value comparison done by the test driver rather in the test because it's easier for a human to locate which test case is failing and in what way.  It's OK for tests to also make the check and print a boolean additionally, but I think they should also print the actual result.  Floating point shouldn't be an issue unless you explicitly print more digits of precision than are reasonable for the type.